includes/ManageWiki.php

```diff
@@ -256,4 +256,59 @@ public static function modifyPermissions( $group, $addp = [], $removep = [], $ad
 			);
 		}
 	}
+
+	public static function defaultCanonicalNamespaces() {
+		global $wgCreateWikiDatabase;
+
+		$dbr = wfGetDB( DB_REPLICA, [], $wgCreateWikiDatabase );
+
+		$res = $dbr->select(
+			'mw_namespaces',
+			'ns_namespace_id',
+			[ 'ns_dbname' => 'default' ]
+		);
+
+		$namespaces = [];
+
+		foreach( $res as $row ) {
+			$namespaces[] = $row->ns_namespace_id;
+		}
+
+		return $namespaces;
+	}
+
+	public static function defaultNamespaces( $namespace ) {
+		global $wgCreateWikiDatabase;
+
+		$dbr = wfGetDB( DB_REPLICA, [], $wgCreateWikiDatabase );
+
+		$res = $dbr->selectRow(
+			'mw_namespaces',
+			[
+				'ns_namespace_name',
+				'ns_searchable',
+				'ns_subpages',
+				'ns_content',
+				'ns_protection',
+				'ns_aliases',
+				'ns_core',
+			],
+			[
+				'ns_dbname' => 'default',
+				'ns_namespace_id' => $namespace
+			]
+		);
+		
+		$ns = [];
+
+		$ns['ns_namespace_name'] = json_decode( $res->ns_namespace_name, true );
```

Comment:
Why are you json_decoding non-json encoded things?

Reply:
Was following how it's done for groups, but don't you need to decode to foreach over the value?

Reply:
Thanks, done.